packages/flutter_tools/lib/src/test/test_compiler.dart

```diff
@@ -168,28 +169,40 @@ class TestCompiler {
       }
 
       Uri? nativeAssetsYaml;
-      final Uri projectUri = FlutterProject.current().directory.uri;
-      final NativeAssetsBuildRunner buildRunner = NativeAssetsBuildRunnerImpl(
-        projectUri,
-        buildInfo.packageConfig,
-        globals.fs,
-        globals.logger,
-      );
-      if (globals.platform.isMacOS) {
-        (nativeAssetsYaml, _) = await buildNativeAssetsMacOS(
-          buildMode: BuildMode.debug,
-          projectUri: projectUri,
-          flutterTester: true,
-          fileSystem: globals.fs,
-          buildRunner: buildRunner,
-      );
+      if (!buildInfo.buildNativeAssets) {
+        nativeAssetsYaml = null;
```

Comment:
Previous reviews explicitly requested to have a final variable that is initialized in every path, so that we don't forget to initialize it in any of the branches.